sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+Each policy has a dedicated purpose and acts on a request or a response or sometimes both.
+When you send a request, the policies are executed in the order that they're added to the pipeline.
+When you receive a response from the service, the policies are executed in reverse order.
+All policies added to the pipeline execute before you send the request and after you receive a response.
+The policy has to decide whether to act on the request, the response, or both.
+For example, a logging policy logs the request and response but the authentication policy is only interested in modifying the request.
+
+The Azure Core framework provides the policy with the necessary request and response data along with any necessary context to execute the policy.
+The policy can then perform its operation with the given data and pass the control along to the next policy in the pipeline.
+
+#### HTTP pipeline policy position
+
+When you make HTTP requests to cloud services, it's important to handle transient failures and to retry failed attempts.
+Because this functionality is a common requirement, Azure Core provides a retry policy that can watch for transient failures and automatically retry the request.
+
+This retry policy, therefore, splits the whole pipeline into two parts:
+policies that execute before the retry policy and policies that execute after the retry policy.
+Policies added before the retry policy execute only once per API operation, and policies added after the retry policy execute as many times as the retries.
+
+So, when building the HTTP pipeline, you should understand whether to execute a policy for each request retry or once per API operation.
+
+![](https://learn.microsoft.com/en-us/azure/developer/java/sdk/media/http-pipeline.svg)
+
+#### Common HTTP pipeline policies
+
+HTTP pipelines for REST-based services have configurations with policies for authentication, retries, logging, telemetry, and specifying the request ID in the header.
+These policies are automatically added into HTTP pipeline.
+
+#### Custom HTTP pipeline policy
+
+The HTTP pipeline policy provides a convenient mechanism to modify or decorate the request and response.
+You can add custom policies to the pipeline that the user or the client library developer created.
+When adding the policy to the pipeline, you can specify whether this policy should be executed per-call or per-retry.
+
+To create a custom HTTP pipeline policy, you just inherit from a base policy class and implement some virtual functions.
+You can then plug the policy into the pipeline.
+
+#### Custom headers in HTTP requests
+
+Below is an example of adding a custom header into each HTTP request.
+The header value is static and doesn't change over time, so we make it a per-operation policy.
+If you want to add some time-variant headers like authentication, you should use per-retry policy.
+
+```C++
+class NewPolicy final : public Azure::Core::Http::Policies::HttpPolicy {
+public:
+  ~NewPolicy() override {}
```

Comment:
Is this necessary?

Reply:
It's useful to ensure the base class (`Azure::Core::Http::Policies::HttpPolicy) has a virtual destructor, so that deleting a derived instance through base pointer is not undefined behavior.

Since we know `Azure::Core::Http::Policies::HttpPolicy` has virtual destructor. The `override` destructor here doesn't make a difference. But I think it's a good practice and want to use it wherever possible.